poker-app/Home.js

```diff
@@ -1,32 +1,33 @@
 import React, { Component } from 'react';
 import { StyleSheet, Text, View, Button, AsyncStorage, TouchableOpacity } from 'react-native';
 import StatsBox from './components/Statsbox'
+const storage = require("./components/AsyncStorageController.js");
 class HomeScreen extends Component {
     state = {
         gamesArray : []
     }
 
     componentDidMount() {
         this.setState({
-            gamesArray: this.retrieveData()
+            gamesArray: storage.retrieveData
         })
     }
 
-    retrieveData = async () => {
-        let keys = await AsyncStorage.getAllKeys()
-        console.log(keys);
-        let games = await AsyncStorage.getItem(keys[0])
-        console.log(JSON.parse(games));
-        return JSON.parse(games);
-    }
+    // retrieveData = async () => {
+    //     let keys = await AsyncStorage.getAllKeys()
+    //     console.log(keys);
+    //     let games = await AsyncStorage.getItem(keys[0])
+    //     console.log(JSON.parse(games));
+    //     return JSON.parse(games);
+    // }
 
     render() {
         return (
             <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
                 <Text>Welcome to Poker</Text>
-                <StatsBox games={this.retrieveData} />
+                <StatsBox games={storage.retrieveData} />
```

Comment:
you call retrieve.data when you have your state?

Reply:
I will send the data from state not call retrieve data again